pkg/operator/ceph/csi/ceph_config.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2022 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/daemon/ceph/client"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	configName = "csi-ceph-conf-override"
+
+	cephConfVal = `[global]
```

Comment:
Are these default settings coming from a downstream requirement? We've never had default ceph.conf settings before, why do we need them now?

Reply:
These are the defaults used in cephcsi from long time.

Reply:
By default, what if we create an empty configmap in rook, and if it's empty then the csi driver sets its default?

Reply:
not possible as the csi driver won't interact with KUbe api's Rook is mounting the configmap as a file to the csi pods. even if the file is empty csi driver cannot write due to a permission issue.